### ISSUE TITLE
Name of grdmath's EXTREMA operator

### ISSUE THREAD

**Issue by anbj** [GenericMappingTools/gmt#7193]
* [grdmath's](https://docs.generic-mapping-tools.org/dev/grdmath.html#operators) EXTREMA gives (from the docs): *Local Extrema: +2/-2 is max/min, +1/-1 is saddle with max/min in x, 0 elsewhere*.
* Only minimum and maximum points are included in the [extrema term](https://en.wikipedia.org/wiki/Maxima_and_minima).
* A term including minimum, maximum *and* saddle point(s) is **critical point**. From [wikipedia](https://en.m.wikipedia.org/wiki/Critical_point_(mathematics)):

  *A critical point (where the function is differentiable) may be either a [local maximum](https://en.m.wikipedia.org/wiki/Local_maximum), a [local minimum](https://en.m.wikipedia.org/wiki/Local_minimum) or a [saddle point](https://en.m.wikipedia.org/wiki/Saddle_point).*

* Would it make sense to change the name of this operator to CRITICALPOINT (or an abbreviation)?

**Comment by PaulWessel** [comment]
It is true that I was a bit quick in naming that function, or perhaps I added saddles later and did not think about that part. **CRITICALPOINT** is a bit long though. I could stomach **CRITICAL**. We will solicit that guru of brevity, Sir @joa-quim for his noble opinion.

**Comment by joa-quim** [comment]
I don't think we should go along that **CRI**(p)**TICAL** path. **EXTREMA** is a very well established term and should not be hidden under a more obscure mathematical definition. On the other hand a ``saddle`` point is not an ``extrema``. Can we have a new operator **SADDLE**? And preferable with a *more comprehensible* explanation? 
This was nearly incomprehensible to me (had to look at the produced grid to confirm)
```
+2/-2 is max/min, +1/-1 is saddle with max/min in x, 0 elsewhere
```

**Comment by PaulWessel** [comment]
We could declare a bug and add **SADDLE** and remove the saddles from **EXTREMA**.

**Comment by joa-quim** [comment]
I thinks that's the best solution.

**Comment by PaulWessel** [comment]
What do we return if we have separate **EXTREMA** and **SADDLE**? Right now we have ±2 for extrema and ±1 for saddle.  Do we switch EXTREMA to also return ±1?

**Comment by joa-quim** [comment]
Do you mean in the same command? That's confusing. I think EXTREMA should do only extrema.
Another question is if EXTREMA should not keep doing also -1/+1 under the hood to maintain backward compat.

**Comment by anbj** [comment]
Can a grid contain characters instead of numbers?

**Comment by PaulWessel** [comment]
I mean should both operators return -1, 0, or +1?

**Comment by joa-quim** [comment]
Ah, preferably they should return the same coding, +-1

**Comment by joa-quim** [comment]
> Can a grid contain characters instead of numbers?

One byte grids, that are actually gray scale *images*, *could* be read by some program as chars (chars are in fact numbers), though first 30 of them in the ASCII table are not visible. But this is divagating.

**Comment by anbj** [comment]
Just a thought; could it instead of -1, 0 and +1 give a (ASCII) letter? Or would that create problems?

**Comment by joa-quim** [comment]
``grdmath`` produces grids. Grids have no letters.

**Comment by anbj** [comment]
This was fixed in #7195.
